src/nvim/message.c

```diff
@@ -2383,6 +2383,15 @@ static void redir_write(char_u *str, int maxlen)
   char_u      *s = str;
   static int cur_col = 0;
 
+  // Append output to capture().
+  if (capture_ga) {
+    int len = maxlen > 0 ? maxlen : (int)STRLEN(str);
+    ga_grow(capture_ga, len);
+    memmove((char *)capture_ga->ga_data
```

Comment:
And it would be best if you added early return

```C
if (maxlen == 0) {
  return;
}
```

at the top. Actually this function does not care if `maxlen == 0` (`*s != NUL` is checked even in this case), though based on documentation it may be a valid argument.